### ISSUE TITLE
E2015 incorrectly raised when type is List<String> and multiple values provided in Default

### ISSUE THREAD

**Issue by verbitan** [aws-cloudformation/cfn-lint#3465]
### CloudFormation Lint Version

cfn-lint 1.5.0

### What operating system are you using?

RHEL 8.9

### Describe the bug

I have a CloudFormation parameter of type `List<String>` with an `AllowedPattern` and `Default` set.

`cfn-lint` validates a single default value correctly, but incorrectly raises `E2015: Default should be allowed by AllowedPattern` when a list of default values is provided.

### Expected behavior

CloudFormation itself accepts this default, so `cfn-lint` should also.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Parameters:
  ListOfCIDR:
    Description: List of CIDR.
    Type: List<String>
    Default: 80.0.0.0/0, 81.0.0.0/8, 82.0.0.0/16
    AllowedPattern: ^([0-9]{1,3}\.){3}[0-9]{1,3}(\/([0-9]|[1-2][0-9]|3[0-2]))?$
```
